pinkie-card/main.qml

```diff
@@ -1,37 +1,57 @@
 import QtQuick 2.7
-import QtQuick.Controls 2.0
-import QtQuick.Layouts 1.0
+import QtQuick.Controls 2.1
+import QtQuick.Layouts 1.3
+import QtQml 2.2
 
 ApplicationWindow {
+    id: root
     visible: true
     width: 640
     height: 480
-    title: qsTr("Hello World")
+    title: qsTr("The Pinkie Card")
 
-    SwipeView {
-        id: swipeView
-        anchors.fill: parent
-        currentIndex: tabBar.currentIndex
+    property date cur_date: new Date()
+    property string cur_lang: 'images/flags/United_States.svg'
 
-        Page1 {
-        }
+    Intro {
+        cur_clock: cur_date.toLocaleDateString() + ", " + cur_date.toLocaleTimeString(Locale.ShortFormat)
+    }
 
-        Page {
-            Label {
-                text: qsTr("Second page")
-                anchors.centerIn: parent
-            }
+    Button {
+        id: lang_select
+        x: 477
+        width: 70
+        height: 36
+        anchors.right: parent.right
+        anchors.rightMargin: 30
+        anchors.top: parent.top
+        anchors.topMargin: 30
+        z: 1
+        onClicked: flag_popup.open()
+
+        background: Image {
+            source: cur_lang
         }
     }
 
-    footer: TabBar {
-        id: tabBar
-        currentIndex: swipeView.currentIndex
-        TabButton {
-            text: qsTr("First")
-        }
-        TabButton {
-            text: qsTr("Second")
+    Popup {
+        id: flag_popup
+        modal: true
+        focus: true
+        // XXX: Should change this awful hack to get the modal centered
```

Comment:
More of a workaround than a hack I'd say. Well looks fine to me anyways.

PS : it's written otherwise, not otherwize :P